rcl/test/CMakeLists.txt

```diff
@@ -139,4 +146,19 @@ function(test_target_function)
   )
 endfunction()
 
+macro(connext_workaround target)
+  if(NOT "${ARGN} " STREQUAL " ")
+    message(FATAL_ERROR "append_link_flags() called with "
+      "unused arguments: ${ARGN}")
+  endif()
+  if("${CMAKE_CXX_COMPILER_ID} " STREQUAL "GNU ")
+    get_target_property(current_link_flags target LINK_FLAGS)
+    set(link_flags "-Wl,--no-as-needed")
+    if(current_link_flags)
+      set(link_flags ${current_link_flags} ${link_flags})
+    endif()
+    set_target_properties(${target} PROPERTIES LINK_FLAGS "${link_flags}")
+  endif()
```

Comment:
Macros should prefix all local variable with an underscore to avoid breaking variables in the parent scope.